homeassistant/components/image_processing/facebox.py

```diff
@@ -173,34 +214,36 @@ def teach(self, name, file_path):
         if (not self.hass.config.is_allowed_path(file_path)
                 or not valid_file_path(file_path)):
             return
-        with open(file_path, 'rb') as open_file:
-            response = requests.post(
-                self._url_teach,
-                data={ATTR_NAME: name, 'id': file_path},
-                files={'file': open_file})
 
-        if response.status_code == 200:
-            self.hass.bus.fire(
-                EVENT_CLASSIFIER_TEACH, {
-                    ATTR_CLASSIFIER: CLASSIFIER,
-                    ATTR_NAME: name,
-                    FILE_PATH: file_path,
-                    'success': True,
-                    'message': None
-                    })
-
-        elif response.status_code == 400:
-            _LOGGER.warning(
-                "%s teaching of file %s failed with message:%s",
-                CLASSIFIER, file_path, response.text)
-            self.hass.bus.fire(
-                EVENT_CLASSIFIER_TEACH, {
-                    ATTR_CLASSIFIER: CLASSIFIER,
-                    ATTR_NAME: name,
-                    FILE_PATH: file_path,
-                    'success': False,
-                    'message': response.text
-                    })
+        response = teach_file(self._url_teach,
+                              self._username,
+                              self._password,
+                              name,
+                              file_path)
+
+        if response is not None:
+            if response.status_code == HTTP_OK:
+                self.hass.bus.fire(
+                    EVENT_CLASSIFIER_TEACH, {
+                        ATTR_CLASSIFIER: CLASSIFIER,
+                        ATTR_NAME: name,
+                        FILE_PATH: file_path,
+                        'success': True,
+                        'message': None
+                        })
+
+            elif response.status_code == HTTP_BAD_REQUEST:
+                _LOGGER.warning(
+                    "%s teaching of file %s failed with message:%s",
+                    CLASSIFIER, file_path, response.text)
+                self.hass.bus.fire(
```

Comment:
What is the benefit of firing an event also for failure?

Reply:
I have an automation to send a notification on teaching, but perhaps I just drop both events (success & fail) here?

Reply:
You know better how the events are useful than me. What is the workflow for teaching you are using or imagining for the future? If we lay that down, it'll probably be more clear what is useful.

Reply:
One of the issues with the facebox free tier is that if 'forgets' faces if you restart the docker container, so I imagine using the service for allowing automations to reteach a folder of faces. I now think that a log message is fine if teaching fails for some reason

Reply:
persistent notification will be better then a log entry :)

Reply:
Have you got an example of notification?

Reply:
hass.components.persistent_notification.create(
            'Error: no model loaded on your facebox container!',
             title=NOTIFICATION_TITLE,
             notification_id=NOTIFICATION_ID)